cogs/factorio.py

```diff
@@ -128,83 +118,99 @@ async def embed_fff(number: int) -> discord.Embed:
+                formatted = linkEx.sub(
+                    lambda m: f"[{m[2] or m[1]}](https://{baseURL}/{m[1]})",
+                    intropar)
+                formatted = formatted.replace("'", "")
+                em = discord.Embed(title=title,
+                    url=url,
+                    description=formatted,
+                    color=discord.Colour.green())
+                if "Infobox" in paragraphs[0]:
+                    image_url = f"https://{baseURL}/images/{title.replace(' ', '_')}.png"
+                    em.set_thumbnail(url=image_url)
+                await bufferMsg.edit(embed=em)
+            else: 
                 engResults = []
-                em = discord.Embed(title=f"Factorio {'' if not stable else 'Stable '}Wiki",
-                                   url=url,
-                                   colour=discord.Colour.gold())
-                for item in soup.find_all("ul", class_="mw-search-results")[0].find_all("li"):
-                    item = item.find_next("div", class_="mw-search-result-heading").find("a")
-                    if langEx.search(item["title"]) is None:
-                        engResults.append(item)
-                if len(engResults) == 1:
-                    item = engResults[0]
-                    await bufferMsg.edit(embed=await wiki_embed(f"https://{baseURL}{item['href']}"))
-                elif len(engResults) > 1:
-                    for item in engResults:
-                        em.add_field(name=item["title"], value=f"[Read More](https://{baseURL}{item['href']})", inline=True)
+                for result in results:
+                    if langEx.search(result["title"]) is None:
+                        engResults.append(result)
+                if engResults != []:
+                    em = discord.Embed(title="Factorio Wiki",
+                                       url=f"https://{baseURL}/index.php?search={searchterm}",
+                                       color=discord.Colour.gold())
+                    for result in engResults:
+                        url = f"https://{baseURL}/{result['title'].replace(' ', '_')}"
+                        em.add_field(name = result["title"], value = f"[Read More]({url})")
                     await bufferMsg.edit(embed=em)
+
                 else:
                     em = discord.Embed(title="Error",
-                                       description=f"Could not find English results for \"{searchterm.title()}\" in {'' if not stable else 'stable '}wiki.",
-                                       colour=discord.Colour.red())
-                    await bufferMsg.edit(embed=em) if ctx.prefix is not None else await bufferMsg.delete()
-            else:
-                await bufferMsg.edit(embed=await wiki_embed(url))
-
+                                       description=f"Could not find English results for \"{searchterm.title()}\" in wiki.",
+                                       color=discord.Colour.red())
 
```

Comment:
```suggestion
                await bufferMsg.edit(embed=await wiki_embed(url))

```
You don't update the embed for the failure message.